lib/spack/spack/cmd/stage.py

```diff
@@ -24,6 +27,10 @@ def setup_parser(subparser):
 
 
 def stage(parser, args):
+    custom_path = os.path.realpath(args.path) if args.path else None
```

Comment:
Is there any reason to call realpath here? I don't think it can hurt here, but in general I think we should use the interface to the filesystem that the user requests, instead of realpathing if it's not absolutely necessary.

Reply:
We need to provide the absolute path when we set `package.path` at the bottom, otherwise it will error when we later change the working directory when setting up the stasge. I will change this to `os.path.abspath()` -- does that address the concern?

Reply:
Converted to `os.path.abspath()`!